### DISCUSSION TITLE
Module not found: Error: Recursion in resolving

### DISCUSSION THREAD

**Discussion by qedric** [hayes0724/shopify-packer#68] [Q&A] [answered]
Hi there,

shopify-packer was working fine until today I created a new repo for another store, and without realising I had installed shopify-packer globally, I installed it again. Now none of my repositories work, I'm really stuck and urgently need to get things runing again! When I run packer start on any repo, I get the following: Any help much appreciated!

`
packer start
Custom webpack configuration found ...\dev.config.js
| Compiling...Failed to compile.

[object Object]

[object Object]

[object Object]

[object Object]

[object Object]

[object Object]

[object Object]

[object Object]

[object Object]

[object Object]

[object Object]

[object Object]

[object Object]

[object Object]

assets by path ../templates/*.liquid 24 bytes 20 assets
assets by path ../snippets/ 6.95 KiB
  asset ..\snippets\script-tags.liquid 3.47 KiB [emitted]
  asset ..\snippets\style-tags.liquid 3.47 KiB [emitted]
  asset ../snippets/.gitkeep 0 bytes [emitted] [from: src/snippets/.gitkeep] [copied]
assets by path ../config/*.json 371 bytes
  asset ../config/settings_schema.json 306 bytes [emitted] [from: src/config/settings_schema.json] [copied]
  asset ../config/settings_data.json 65 bytes [emitted] [from: src/config/settings_data.json] [copied]
asset runtime.js 37.5 KiB [emitted] (name: runtime)
asset ../layout/theme.liquid 631 bytes [emitted] [from: src/layout/theme.liquid] [copied]
asset ../locales/en.default.json 3 bytes [emitted] [from: src/locales/en.default.json] [copied]
asset ../sections/.gitkeep 0 bytes [emitted] [from: src/sections/.gitkeep] [copied]
asset .gitkeep 0 bytes [emitted] [from: src/assets/.gitkeep] [copied]
runtime modules 24.7 KiB 8 modules

ERROR in layout.theme
Module not found: Error: Recursion in resolving
Stack:
  undefined: (<path to repo removed>) <path to use removed>\AppData\Roaming\nvm\v15.6.0\node_modules\@hayes0724\shopify-packer\src\webpack\hot-client.js

ERROR in layout.theme
Module not found: Error: Recursion in resolving
Stack:
  undefined: (<path to repo removed>) <path to repo removed>\src\scripts\layout\theme.js

ERROR in template.collection
Module not found: Error: Recursion in resolving
Stack:
  undefined: (<path to repo removed>) <path to user folder removed>\AppData\Roaming\nvm\v15.6.0\node_modules\@hayes0724\shopify-packer\src\webpack\hot-client.js

ERROR in template.collection
Module not found: Error: Recursion in resolving
Stack:
  undefined: (<path to repo removed>) <path to repo removed>\src\scripts\templates\collection.js

ERROR in template.index
Module not found: Error: Recursion in resolving
Stack:
  undefined: (<path to repo removed>) <path to user folder removed>\AppData\Roaming\nvm\v15.6.0\node_modules\@hayes0724\shopify-packer\src\webpack\hot-client.js

ERROR in template.index
Module not found: Error: Recursion in resolving
Stack:
  undefined: (<path to repo removed>) <path to repo removed>\src\scripts\templates\index.js

ERROR in template.login
Module not found: Error: Recursion in resolving
Stack:
  undefined: (<path to repo removed>) <path to user folder removed>\AppData\Roaming\nvm\v15.6.0\node_modules\@hayes0724\shopify-packer\src\webpack\hot-client.js

ERROR in template.login
Module not found: Error: Recursion in resolving
Stack:
  undefined: (<path to repo removed>) <path to repo removed>\src\scripts\templates\customers\login.js

ERROR in template.product
Module not found: Error: Recursion in resolving
Stack:
  undefined: (<path to repo removed>) <path to user folder removed>\AppData\Roaming\nvm\v15.6.0\node_modules\@hayes0724\shopify-packer\src\webpack\hot-client.js

ERROR in template.product
Module not found: Error: Recursion in resolving
Stack:
  undefined: (<path to repo removed>) <path to repo removed>\src\scripts\templates\product.js

ERROR in template.register
Module not found: Error: Recursion in resolving
Stack:
  undefined: (<path to repo removed>) <path to user folder removed>\AppData\Roaming\nvm\v15.6.0\node_modules\@hayes0724\shopify-packer\src\webpack\hot-client.js

ERROR in template.register
Module not found: Error: Recursion in resolving
Stack:
  undefined: (<path to repo removed>) <path to repo removed>\src\scripts\templates\customers\register.js

ERROR in   Error: Child compilation failed:
  Module not found: Error: Recursion in resolving
  Stack:
    undefined: (<path to repo removed>) <path to user folder removed>\AppData\Roaming\nvm\v15.6.0\node_modules\@hayes0724\shop  ify-packer\node_modules\html-webpack-plugin\lib\loader.js
  ModuleNotFoundError: Module not found: Error: Recursion in resolving
  Stack:
    undefined: (<path to repo removed>) <path to user folder removed>\AppData\Roaming\nvm\v15.6.0\node_modules\@hayes0724\shop  ify-packer\node_modules\html-webpack-plugin\lib\loader.js

  - Compilation.js:1668
    [v15.6.0]/[shopify-packer]/[webpack]/lib/Compilation.js:1668:28

  - NormalModuleFactory.js:712
    [v15.6.0]/[shopify-packer]/[webpack]/lib/NormalModuleFactory.js:712:13

  - NormalModuleFactory.js:273
    [v15.6.0]/[shopify-packer]/[webpack]/lib/NormalModuleFactory.js:273:22

  - NormalModuleFactory.js:402
    [v15.6.0]/[shopify-packer]/[webpack]/lib/NormalModuleFactory.js:402:22

  - NormalModuleFactory.js:121
    [v15.6.0]/[shopify-packer]/[webpack]/lib/NormalModuleFactory.js:121:11

  - NormalModuleFactory.js:576
    [v15.6.0]/[shopify-packer]/[webpack]/lib/NormalModuleFactory.js:576:23

  - async.js:2830
    [v15.6.0]/[shopify-packer]/[neo-async]/async.js:2830:7

  - async.js:2925 done
    [v15.6.0]/[shopify-packer]/[neo-async]/async.js:2925:13

  - Module not found: Error: Recursion in resolving

  - Stack:

  - undefined: (<path to repo removed>) <path to user folder removed>\AppData\Roaming\nvm\v15.6.0\node_modules\@hayes0724\shop    ify-packer\node_modules\html-webpack-plugin\lib\loader.js

  - ModuleNotFoundError: Module not found: Error: Recursion in resolving

  - Stack:

  - undefined: (<path to repo removed>) <path to user folder removed>\AppData\Roaming\nvm\v15.6.0\node_modules\@hayes0724\shop    ify-packer\node_modules\html-webpack-plugin\lib\loader.js

  - Compilation.js:1668
    [v15.6.0]/[shopify-packer]/[webpack]/lib/Compilation.js:1668:28

  - NormalModuleFactory.js:712
    [v15.6.0]/[shopify-packer]/[webpack]/lib/NormalModuleFactory.js:712:13

  - NormalModuleFactory.js:273
    [v15.6.0]/[shopify-packer]/[webpack]/lib/NormalModuleFactory.js:273:22

  - NormalModuleFactory.js:402
    [v15.6.0]/[shopify-packer]/[webpack]/lib/NormalModuleFactory.js:402:22

  - NormalModuleFactory.js:121
    [v15.6.0]/[shopify-packer]/[webpack]/lib/NormalModuleFactory.js:121:11

  - NormalModuleFactory.js:576
    [v15.6.0]/[shopify-packer]/[webpack]/lib/NormalModuleFactory.js:576:23

  - async.js:2830
    [v15.6.0]/[shopify-packer]/[neo-async]/async.js:2830:7

  - async.js:2925 done
    [v15.6.0]/[shopify-packer]/[neo-async]/async.js:2925:13

  - child-compiler.js:169
    [v15.6.0]/[shopify-packer]/[html-webpack-plugin]/lib/child-compiler.js:169:18

  - Compiler.js:534
    [v15.6.0]/[shopify-packer]/[webpack]/lib/Compiler.js:534:11

  - Compiler.js:1087
    [v15.6.0]/[shopify-packer]/[webpack]/lib/Compiler.js:1087:17

  - Hook.js:154 AsyncSeriesHook.lazyCompileHook
    [v15.6.0]/[shopify-packer]/[webpack]/[tapable]/lib/Hook.js:154:20

  - Compiler.js:1083
    [v15.6.0]/[shopify-packer]/[webpack]/lib/Compiler.js:1083:33

  - Compilation.js:2424
    [v15.6.0]/[shopify-packer]/[webpack]/lib/Compilation.js:2424:11

  - Hook.js:154 AsyncSeriesHook.lazyCompileHook
    [v15.6.0]/[shopify-packer]/[webpack]/[tapable]/lib/Hook.js:154:20

  - Compilation.js:2417
    [v15.6.0]/[shopify-packer]/[webpack]/lib/Compilation.js:2417:38

ERROR in   Error: Child compilation failed:
  Module not found: Error: Recursion in resolving
  Stack:
    undefined: (<path to repo removed>) <path to user folder removed>\AppData\Roaming\nvm\v15.6.0\node_modules\@hayes0724\shop  ify-packer\node_modules\html-webpack-plugin\lib\loader.js
  ModuleNotFoundError: Module not found: Error: Recursion in resolving
  Stack:
    undefined: (<path to repo removed>) <path to user folder removed>\AppData\Roaming\nvm\v15.6.0\node_modules\@hayes0724\shop  ify-packer\node_modules\html-webpack-plugin\lib\loader.js

  - Compilation.js:1668
    [v15.6.0]/[shopify-packer]/[webpack]/lib/Compilation.js:1668:28

  - NormalModuleFactory.js:712
    [v15.6.0]/[shopify-packer]/[webpack]/lib/NormalModuleFactory.js:712:13

  - NormalModuleFactory.js:273
    [v15.6.0]/[shopify-packer]/[webpack]/lib/NormalModuleFactory.js:273:22

  - NormalModuleFactory.js:402
    [v15.6.0]/[shopify-packer]/[webpack]/lib/NormalModuleFactory.js:402:22

  - NormalModuleFactory.js:121
    [v15.6.0]/[shopify-packer]/[webpack]/lib/NormalModuleFactory.js:121:11

  - NormalModuleFactory.js:576
    [v15.6.0]/[shopify-packer]/[webpack]/lib/NormalModuleFactory.js:576:23

  - async.js:2830
    [v15.6.0]/[shopify-packer]/[neo-async]/async.js:2830:7

  - async.js:2925 done
    [v15.6.0]/[shopify-packer]/[neo-async]/async.js:2925:13

  - Module not found: Error: Recursion in resolving

  - Stack:

  - undefined: (<path to repo removed>) <path to user folder removed>\AppData\Roaming\nvm\v15.6.0\node_modules\@hayes0724\shop    ify-packer\node_modules\html-webpack-plugin\lib\loader.js

  - ModuleNotFoundError: Module not found: Error: Recursion in resolving

  - Stack:

  - undefined: (<path to repo removed>) <path to user folder removed>\AppData\Roaming\nvm\v15.6.0\node_modules\@hayes0724\shop    ify-packer\node_modules\html-webpack-plugin\lib\loader.js

  - Compilation.js:1668
    [v15.6.0]/[shopify-packer]/[webpack]/lib/Compilation.js:1668:28

  - NormalModuleFactory.js:712
    [v15.6.0]/[shopify-packer]/[webpack]/lib/NormalModuleFactory.js:712:13

  - NormalModuleFactory.js:273
    [v15.6.0]/[shopify-packer]/[webpack]/lib/NormalModuleFactory.js:273:22

  - NormalModuleFactory.js:402
    [v15.6.0]/[shopify-packer]/[webpack]/lib/NormalModuleFactory.js:402:22

  - NormalModuleFactory.js:121
    [v15.6.0]/[shopify-packer]/[webpack]/lib/NormalModuleFactory.js:121:11

  - NormalModuleFactory.js:576
    [v15.6.0]/[shopify-packer]/[webpack]/lib/NormalModuleFactory.js:576:23

  - async.js:2830
    [v15.6.0]/[shopify-packer]/[neo-async]/async.js:2830:7

  - async.js:2925 done
    [v15.6.0]/[shopify-packer]/[neo-async]/async.js:2925:13

  - child-compiler.js:169
    [v15.6.0]/[shopify-packer]/[html-webpack-plugin]/lib/child-compiler.js:169:18

  - Compiler.js:534
    [v15.6.0]/[shopify-packer]/[webpack]/lib/Compiler.js:534:11

  - Compiler.js:1087
    [v15.6.0]/[shopify-packer]/[webpack]/lib/Compiler.js:1087:17

  - Hook.js:154 AsyncSeriesHook.lazyCompileHook
    [v15.6.0]/[shopify-packer]/[webpack]/[tapable]/lib/Hook.js:154:20

  - Compiler.js:1083
    [v15.6.0]/[shopify-packer]/[webpack]/lib/Compiler.js:1083:33

  - Compilation.js:2424
    [v15.6.0]/[shopify-packer]/[webpack]/lib/Compilation.js:2424:11

  - Hook.js:154 AsyncSeriesHook.lazyCompileHook
    [v15.6.0]/[shopify-packer]/[webpack]/[tapable]/lib/Hook.js:154:20

  - Compilation.js:2417
    [v15.6.0]/[shopify-packer]/[webpack]/lib/Compilation.js:2417:38

2 ERRORS in child compilations (Use 'stats.children: true' resp. '--stats-children' for more details)
webpack 5.28.0 compiled with 16 errors in 585 ms
| Compiling...
↑  Uploading to Shopify...
i Deploy Mode: replace
i Environment: development

/ Compiling...Terminate batch job (Y/N)?
Terminate batch job (Y/N)? y
`

**Comment by hayes0724** [comment, marked as answer]
Hi,

It would help to see your package.json, packer version and what os that your using.

looking at the stack trace it seems it's having issues finding or loading (possible permission issue) the hot-client.js file. It's a simple file that is added to all dev entrypoints in development mode (watch/start)

```
/* eslint-env browser */
const tmpPublicPath = __webpack_public_path__;
__webpack_public_path__ = __webpack_public_path__.replace(/\/$/, '');
// eslint-disable-next-line node/no-missing-require
const client = require('webpack-hot-middleware/client?dynamicPublicPath=true');
__webpack_public_path__ = tmpPublicPath;
window.__shopify_should_reload__ = false;

client.subscribe((event) => {
  if (event.action === 'shopify_upload_finished') {
    if (event.force || window.__shopify_should_reload__) {
      window.location.reload();
    }
  }
});
```
Potential fixes:
1. removing packer globally and reinstalling
2. installing a different version in the project. You can have a global version and a local override. The local will always be used and can be a different version. 

Packer v1 - uses webpack 4 and the latest version is v1.4.0
Packer v2 - uses webpack 5 and has a ton of optimizations.

**Reply by qedric** [reply]
Many thanks! I uninstalled all global and local packages and started again with a single global packer install - all working nicely again

**Comment by hayes0724** [comment]
Just noticed another thing latest version of packer is using: `5.27.1` and the output above is showing webpack `5.28.0`
